src/main/scala/io/iohk/ethereum/vm/EvmConfig.scala

```diff
@@ -0,0 +1,177 @@
+package io.iohk.ethereum.vm
+
+import io.iohk.ethereum.utils.Config
+
+// scalastyle:off number.of.methods
+// scalastyle:off number.of.types
+// scalastyle:off magic.number
+object EvmConfig {
+
+  /**
+    * returns the evm config that should be used for given block
+    */
+  def forBlock(blockNumber: BigInt): EvmConfig = {
+    // highest transition block that is less/equal to `blockNumber`
+    transitionBlockToConfigMapping
+      .filterKeys(_ <= blockNumber)
+      .maxBy(_._1)
+      ._2
+  }
+
+  val FrontierConfig = EvmConfig(
+    feeSchedule = FeeSchedule.FrontierFeeSchedule,
+    opCodes = OpCodes.FrontierOpCodes,
+    exceptionalFailedCodeDeposit = false)
+
+  /*
+    TODO (CREATE):
+    If contract creation does not have enough gas to pay for the final gas fee
+    for adding the contract code to the state, the contract creation fails (ie. goes out-of-gas)
+    rather than leaving an empty contract.
+
+    See: exceptional_failed_code_deposit in Parity
+   */
+  val HomesteadConfig = EvmConfig(
+    feeSchedule = FeeSchedule.HomesteadFeeSchedule,
+    opCodes = OpCodes.HomesteadOpCodes,
+    exceptionalFailedCodeDeposit = true)
+
+  /*
+  TODO(CREATE): sub_gas_cap_divisor
+    If Some(x): let limit = GAS * (x - 1) / x; let CALL's gas = min(requested, limit). let CREATE's gas = limit.
+    If None: let CALL's gas = (requested > GAS ? [OOG] : GAS). let CREATE's gas = GAS
+   */
+  val PostEIP150Config = HomesteadConfig.copy(
+    feeSchedule = FeeSchedule.PostEIP150FeeSchedule,
+    subGasCapDivisor = Some(64))
+
+  val PostEIP160Config = PostEIP150Config.copy(
+    feeSchedule = FeeSchedule.PostEIP160FeeSchedule)
+
+  private val transitionBlockToConfigMapping: Map[BigInt, EvmConfig] = Map(
+    Config.Blockchain.frontierBlockNumber -> FrontierConfig,
+    Config.Blockchain.homesteadBlockNumber -> HomesteadConfig,
+    Config.Blockchain.eip150BlockNumber -> PostEIP150Config,
+    Config.Blockchain.eip160BlockNumber -> PostEIP160Config)
+
+}
+
+case class EvmConfig(
+    feeSchedule: FeeSchedule,
+    opCodes: List[OpCode],
+    exceptionalFailedCodeDeposit: Boolean,
+    maxMemory: UInt256 = UInt256(Long.MaxValue), /* used to artificially limit memory usage by incurring maximum gas cost */
+    maxCallDepth: Int = 1024,
+    subGasCapDivisor: Option[Long] = None) {
```

Comment:
As we know, just because it's defined in YP doesn't mean it won't change :) Even though I also think it won't change in the nearest future, I'd still leave that configurable